docs/DeveloperGuide.md

```diff
@@ -321,11 +325,186 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+2.  System shows a list of persons
+3.  User requests to delete a contact
+4.  System gives a prompt to confirm whether the user wants to delete the contact
+5.  User confirms they want to delete the contact
+6.  System deletes the contact and shows the updated list of persons
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. System shows an error message prompting the user to put in a valid index.
+
+      Use case resumes at step 2.
+
+* 4a. User says they do not want to delete the contact.
+
+    * 4a1. System shows a message indicating the contact was not deleted.
+
+      Use case ends.
+
+**Use case: UCSH04 Receive a prompt when clearing the system**
+
+**MSS**
+
+1. User requests to clear the system of all persons
+2. System gives a prompt to confirm whether the user wants to clear all contacts
+3. User confirms they want to clear all contacts
+4. System deletes all contacts and shows a blank list of persons
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. User says they do not want to clear all their contacts.
+
+    * 2a1. System shows a message indicating the system was not cleared.
+
+      Use case ends.
+
+
+**Use case: UCSH05 Assign dietary requirement to contact**
+
+**MSS**
+
+1.  User requests to list persons
+2.  System shows a list of persons
+3.  User requests to add a dietary status to the person
+4.  System adds the dietary status to the contact and shows list of persons with new details
+    
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. System shows an error message prompting the user to put in a valid index.
+
+      Use case resumes at step 2.
+
+**Use case: UCSH06 Sort contacts in alphabetical order**
+
+**MSS**
+
+1.  User requests to show a list of persons sorted alphabetically
+2.  System shows the list of persons sorted in alphabetical order
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. The list is empty.
+
+  Use case ends.
+
+**Use case: UCSH07 Add additional information for a person**
+
+**MSS**
+
+1.  User requests to list persons
+2.  System shows a list of persons
+3.  User requests to add additional information for a person
+4.  System adds the additional information to the contact and shows list of persons with new details
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. System shows an error message prompting the user to put in a valid index.
+
+      Use case resumes at step 2.
+
+* 3a. The additional information is blank.
+
+    * 3a1. System shows an error message prompting the user to type in the additional information.
+
+      Use case resumes at step 2.
+
+**Use case: UCSH08 See sample contacts in the system before starting to modify it**
+
+Preconditions: User has not added or modified the system previously
```

Comment:
Perhaps can make it clearer, "User has not added contacts"

Reply:
Will do